db/python/tables/sample.py

```diff
@@ -687,3 +693,55 @@ async def get_samples_with_missing_participants_by_internal_id(
             )
         )
         return samples
+
+    async def get_monthly_samples_count_per_project(
+        self, project_ids: list[int] | None = None
+    ) -> dict[int, dict[date, int]]:
+        """
+        Get a map of {project_id: {month:count} for list of project_ids
+        If project_ids is empty, return all projects
+        """
+
+        where_str = 'WHERE project in :project_ids' if project_ids else ''
+        values = {'project_ids': project_ids} if project_ids else {}
+
+        _query = f"""
+        WITH t AS(
+            SELECT project, id, min(row_start) as sample_first_date
+            FROM sample FOR SYSTEM_TIME ALL
+            {where_str}
+            GROUP BY project, id
+        )
+        SELECT project, CAST(EXTRACT(YEAR FROM sample_first_date) AS INTEGER) AS year, CAST(EXTRACT(MONTH FROM sample_first_date) AS INTEGER) AS month, COUNT(*) AS count
```

Comment:
would be good to add some newlines here for readability

Reply:
Good point, had addressed it.